src/java-opentimelineio/src/main/java/io/opentimeline/opentimelineio/AnyVector.java

```diff
@@ -0,0 +1,232 @@
+package io.opentimeline.opentimelineio;
+
+import io.opentimeline.OTIONative;
+import io.opentimeline.OTIOObject;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.TreeSet;
+import java.util.function.Consumer;
+
+public class AnyVector extends OTIOObject implements Collection<Any> {
+
+    public AnyVector() {
+        this.initObject();
+    }
+
+    AnyVector(OTIONative otioNative) {
+        this.nativeManager = otioNative;
+    }
+
+    private void initObject() {
+        this.initialize();
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private native void initialize();
+
+    public class Iterator extends OTIOObject implements java.util.Iterator<Any> {
+        private boolean startedIterating = false;
+
+        private Iterator(AnyVector anyVector) {
+            this.initObject(anyVector);
+        }
+
+        Iterator(OTIONative otioNative) {
+            this.nativeManager = otioNative;
+        }
+
+        private void initObject(AnyVector anyVector) {
+            this.initialize(anyVector);
+            this.nativeManager.className = this.getClass().getCanonicalName();
+        }
+
+        private native void initialize(AnyVector anyVector);
+
+        public boolean hasNext() {
+            if (!startedIterating && size() > 0) return true;
+            return hasNextNative(AnyVector.this);
+        }
+
+        public boolean hasPrevious() {
+            return hasPreviousNative(AnyVector.this);
+        }
+
+        public Any next() {
+            startedIterating = true;
+            return nextNative(AnyVector.this);
+        }
+
+        public Any previous() {
+            return previousNative(AnyVector.this);
+        }
+
+        private native Any nextNative(AnyVector anyVector);
+
+        private native Any previousNative(AnyVector anyVector);
+
+        private native boolean hasNextNative(AnyVector anyVector);
+
+        private native boolean hasPreviousNative(AnyVector anyVector);
+    }
+
+    public AnyVector.Iterator iterator() {
+        return new AnyVector.Iterator(this);
+    }
+
+    @Override
+    public Object[] toArray() {
+        return getArray();
+    }
+
+    @Override
+    public <T> T[] toArray(T[] ts) {
+        if (ts instanceof Any[])
+            return (T[]) getArray();
+        throw new ClassCastException();
+    }
+
+    private native Any[] getArray();
+
+    public native Any get(int index);
+
+    public native boolean add(Any any);
+
+    @Override
+    public boolean remove(Object o) {
```

Comment:
just curious if C++ does not have remove, is that why it is the only functional non-native method here?